File: backend/src/main/java/com/backend/domain/userPenalty/repository/UserPenaltyRepositoryCustom.java
package com.backend.domain.userPenalty.repository;

import com.backend.domain.userPenalty.entity.UserPenalty;
import java.time.LocalDateTime;
import java.util.List;

public interface UserPenaltyRepositoryCustom {

    // 🔹 만료된 정지 내역 (자동 해제용)
    List<UserPenalty> findExpiredPenalties(LocalDateTime now);
}

File: backend/src/main/java/com/backend/domain/userPenalty/repository/UserPenaltyRepositoryCustom.kt
package com.backend.domain.userPenalty.repository

import com.backend.domain.userPenalty.entity.UserPenalty
import java.time.LocalDateTime

interface UserPenaltyRepositoryCustom {

    // 만료된 정지 내역 조회 (자동 해제용)
    fun findExpiredPenalties(now: LocalDateTime): List<UserPenalty>
}
